rpc/connection_log.go

```diff
@@ -99,25 +119,37 @@ func (l *connectionLogLogrus) logSkip(fields ...logField) LogrusEntry {
 		mFields[f.key] = f.value
 	}
 	mFields["section"] = l.section
+	mFields["identifier"] = l.randBytes
 	mFields["file"], mFields["line"] = file, line
-	return l.log.WithFields(mFields)
+
+	if msgI, ok := mFields[msgKey]; ok {
+		if msg, ok = msgI.(string); !ok {
+			msg = fmt.Sprintf("%s", msgI)
+		}
+	} else {
+		// If "msg" isn't present in fields, just use a formatted string. It
+		// won't index well but it's better than losing information.
+		msg = formatLogFields(format, fields...)
+	}
+
+	return l.log.WithFields(mFields), msg
 }
 
-func (l *connectionLogLogrus) Warning(
-	msg string, format string, fields ...logField) {
-	l.logSkip(fields...).Warning(msg)
+func (l *connectionLogLogrus) Warning(format string, fields ...logField) {
+	e, msg := l.getLogEntryAndExtractMsg(format, fields)
+	e.Warning(msg)
 }
 
-func (l *connectionLogLogrus) Debug(
-	msg string, format string, fields ...logField) {
+func (l *connectionLogLogrus) Debug(format string, fields ...logField) {
 	// Nasty hack to lift Debug to Info: we can't just do Info everywhere in
 	// this package since they'll leak to CLI in keybase/client. We also can't
 	// just turn on Debug logging on server-side. Since logrus loggers are only
 	// used by server side, we simply override Debug with Info here.
-	l.logSkip(fields...).Info(msg)
+	e, msg := l.getLogEntryAndExtractMsg(format, fields)
```

Comment:
Could just call `l.Info()` here instead of duplicating the code, right?

Reply:
Good idea will do!